src/fc/qemu/hazmat/qemu.py

```diff
@@ -418,6 +419,32 @@ def write_file(self, path, content: bytes):
         finally:
             self.guestagent.cmd("guest-file-close", handle=handle)
 
+    def exec(self, executable: str, arg: List[str]):
+        output = self.guestagent.cmd(
+            "guest-exec",
+            path=executable,
+            arg=arg,
+        )
+        try:
+            pid = output["pid"]
+        except KeyError:
+            raise RuntimeError("Command could not be started. No PID")
+        timeout = TimeOut(5, 1, raise_on_timeout=True)
+        while timeout.tick():
+            status = self.guestagent.cmd("guest-exec-status", pid=pid)
+            if status["exited"] is True:
```

Comment:
slight refactoring for readability:

* turn the `if exited` into a guarding
* use `dict.get()` default to avoid the additional conditions layer